stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/CSS.java

```diff
@@ -136,57 +112,56 @@ public void run() {
             elementForNodeId,
             new StyleAccumulator() {
               @Override
-              public void store(String name, String value, boolean isDefault) {
+              public void store(String selector, String name, String value, boolean isDefault) {
                 if (!isDefault) {
-                  CSSProperty property = new CSSProperty();
-                  property.name = name;
-                  property.value = value;
+                  ArrayList<CSSProperty> properties = mSelectorCollector.get(selector);
+                  if (properties == null) {
+                    properties = new ArrayList<>();
+                    mSelectorCollector.put(selector, properties);
+                  }
 
-                  match.rule.style.cssProperties.add(property);
-                }
-              }
-            });
-
-        mDocument.getElementAccessibilityStyles(
-            elementForNodeId,
-            new StyleAccumulator() {
-              @Override
-              public void store(String name, String value, boolean isDefault) {
-                if (!isDefault) {
-                  CSSProperty property = new CSSProperty();
+                  final CSSProperty property = new CSSProperty();
                   property.name = name;
                   property.value = value;
-
-                  accessibilityMatch.rule.style.cssProperties.add(property);
+                  properties.add(property);
                 }
               }
             });
       }
     });
 
+    final GetMatchedStylesForNodeResult result = new GetMatchedStylesForNodeResult();
+    result.matchedCSSRules = new ArrayList<>();
     result.inherited = Collections.emptyList();
     result.pseudoElements = Collections.emptyList();
 
-    return result;
-  }
+    for (String selectorName : mSelectorCollector.keySet()) {
+      final ArrayList<CSSProperty> properties = mSelectorCollector.get(selectorName);
```

Comment:
is there any reason we're iterating over the `keySet()` and then retrieving the value via `get()`, versus just iterating over the `entrySet()` and using `Map.Entry#getKey()` and `#getValue()` ? they're functionally equivalent, although the latter (iterating over entrySet) avoids an extra hashtable lookup ... which isn't very expensive ... 

iterating the `keySet()` would make perfect sense if we're using an implementation of `Map` that creates the `Map.Entry`s on-demand, such as `ArrayMap` or (iirc) `IdentityHashMap`, but `HashMap` creates them up-front